Cabal/Distribution/Compat/ReadP.hs

```diff
@@ -297,6 +298,11 @@ skipSpaces =
   skip (c:s) | isSpace c = do _ <- get; skip s
   skip _                 = do return ()
 
+skipSpaces1 :: ReadP r ()
+-- ^ Like 'skipSpaces' but succeeds only if there is at least one
```

Comment:
@robjhen In general, Compat modules exist to work around time-limited problems in base libraries. So at some point in time in the past (git log to find out) ReadP was not available in the base package for versions of base we wanted to support. That version may have gone out of our support window, in which case it's OK to drop the module!